docs/life-of-a-request.md

```diff
@@ -0,0 +1,200 @@
+# Life of a Request
+
+It can sometimes be confusing to figure out how to write a new feature,
+or debug an existing one. Let's try to follow a request through the
+Zulip codebase, and dive deep into how each part works.
+
+## A request is sent to the server, and handled by Nginx
+
+Everything in Zulip goes through nginx, when deployed in production.
```

Comment:
Maybe `When Zulip is deployed in production, everything goes through nginx.` to avoid the unclear antecedent.